### PR TITLE
refactor(zones): migrate components to composition API

### PR DESCRIPTION
**Pull request by kleinfreund**
Migrates all zone-related (and a few simple ones) to composition API.

This aids in providing better type system support for component code which is lackluster for components written in Options API style.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: src/app/zones/views/__snapshots__/ZonesView.spec.ts.snap [hunk 1]
```diff
@@ -888,6 +888,30 @@ exports[`ZonesView.vue renders snapshot when multizone 1`] = `
...
+                                    >
+                                      
+                                      online
+                                      
+                                    </div>
+                                  </div>
+                                  <!---->
+                                </div>
+                              </p>
+                            </li>
```
Comment (johncowen) [inline src/app/zones/views/__snapshots__/ZonesView.spec.ts.snap:914]: We gained a `status` output here, but I don't see any mocks that changed, so whilst we might need to show `status` I'm guessing some logic changed somewhere to mean that this now shows?

**Review by johncowen — APPROVED** [review]
I had one question just incase, but I think even that is fine, if it is then 👍

File: src/app/zones/views/__snapshots__/ZonesView.spec.ts.snap (hunk 1 above)
Comment (kleinfreund) [reply src/app/zones/views/__snapshots__/ZonesView.spec.ts.snap:914]: That’s right. The component clearly indicated that the status was meant to be shown (hence the table column for it). There was other code surrounding statuses, too. However, this was partially broken and resulted in the statuses never making it into the page. Specifically, the component’s `getEntity` method didn’t add the status to the `entity` object and so the table, while trying to display it, never had a value.

So, technically that’s a small bug fix.

Comment (johncowen) [reply src/app/zones/views/__snapshots__/ZonesView.spec.ts.snap:914]: Gotcha, merge away!

**Pull request merged**